### PR TITLE
Add detail to env var and CI info

### PR DESCRIPTION
**Pull request by meiqimichelle**
Team Test Analytics got feedback on this section of our docs via a beta particpant Slack channel. Grant and I hopped on a screenshare to offer some clarifications for this section of our documentation.

### CODE REVIEW

**Review by plaindocs — APPROVED** [review]
There is so much of my change here that :+1: feels pretty meaningless, but here it is. Be good to have someone that isn't @meiqimichelle or myself give it a :+1: before merging.

**Comment by JuanitoFatas** [comment]
@meiqimichelle This Pull Request renders like

localhost_3000_docs_test-analytics_integrations

**Comment by meiqimichelle** [comment]
@JuanitoFatas that's not what I'm seeing locally when I pull it down. But no worries -- earlier when I was asking about this in our handout time, I was just being lazy, and wondered if there was a link to Heroku for docs I was missing (and there was! now I know). Thank you for the screenshot tho!

@plaindocs it would be best to get someone else to review as well, but I do think this is a huge step in the right direction compared to what is live today (aka maybe we should 🚢 and improve again later).

I just read it over -- the only thing that I noticed is that we say:

`The only required environment variable is BUILDKITE_ANALYTICS_KEY, but some features of Test Analytics aren't available if the other environment variables are not set:`

...but then don't refer to this env var in any of the tables. I remember that we talked about this env var being called KEY, but the line the table called key actually mapping to something else.....maybe......but I don't remember, then, where we describe what BUILDKITE_ANALYTICS_KEY is to begin with. Does that need to be clarified somewhere? (Or perhaps if you're more across how env vars work than I am, this is clear already.)

**Comment by plaindocs** [comment]
@JuanitoFatas yeah, that preview pic looks like before I made the big updatse.

Heroku previews are done at the mo, due to the big OAuth thing, but @lachlanhardy is looking into it.

@meiqimichelle good catch, I'll fix!

**Comment by plaindocs** [comment]
@meiqimichelle  just to confirm, 

`key` = `BUILDKITE_ANALYTICS_KEY`  (automatically set to =  BUILDKITE_BUILD_ID on Buildkite), so I do need to reword that part.

File: pages/test_analytics/integrations.md.erb [hunk 1]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
 ]
 ```
 
+## Environment variables
+
+Test Analytics uses environment variables to populate test information. Whichever CI provider you use to run your builds, you need to make sure that environment variables are available to the RSpec integration or import API, including making sure that they are passed through to a container if your builds are running in containers.
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:341]:
```suggestion
Test Analytics uses environment variables to populate test information. Regardless of the CI provider you use to run your builds, you need to make sure that environment variables are available to the RSpec integration or import API. You also need to make sure that the environment variables are passed through to a container if your builds are running in containers.
```

File: pages/test_analytics/integrations.md.erb [hunk 2]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+
+* [RSpec on Buildkite](#environment-variables-rspec-on-buildkite)
+* [RSpec on CircleCI](#environment-variables-rspec-on-circleci)
+* [RSpec on GitHub Actions](#environment-variables-rspec-on-github-actions)
+* [RSpec on other CI providers](#environment-variables-other-ci-providers-and-integrations)
+* [Other CI providers and integrations](#environment-variables-other-ci-providers-and-integrations)
+
+The only required environment variable is the unique key for the build that initiated the Test Analytics run (`BUILDKITE_ANALYTICS_KEY`).
+
+If you're not using CI, you can still use Test Analytics by running an integration manually.
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:353]: Links? A little more info?..

File: pages/test_analytics/integrations.md.erb [hunk 3]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+
+### RSpec on Buildkite
+
+The following variables are set automatically when the RSpec integration is running on Buildkite:
+
+| Variable                       | Description and source                                                                   |
+|--------------------------------|------------------------------------------------------------------------------------------|
+| `BUILDKITE_ANALYTICS_KEY`      | Unique key for the build that initiated the Test Analytics run<br />`BUILDKITE_BUILD_ID` |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:361]:
```suggestion
| `BUILDKITE_ANALYTICS_KEY`      | a unique key for the build that initiated the Test Analytics run<br />`BUILDKITE_BUILD_ID` |
```

File: pages/test_analytics/integrations.md.erb [hunk 4]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `BUILDKITE_ANALYTICS_URL`      | URL that links to the build <br /> `BUILDKITE_BUILD_URL`                                 |
+| `BUILDKITE_ANALYTICS_BRANCH`   | the branch or ref that this run is for<br />`BUILDKITE_BRANCH`                           |
+| `BUILDKITE_ANALYTICS_SHA`      | the commit sha for the head of the branch<br />`BUILDKITE_COMMIT`                        |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:364]:
```suggestion
| `BUILDKITE_ANALYTICS_SHA`      | the commit SHA for the head of the branch<br />`BUILDKITE_COMMIT`                        |
```

File: pages/test_analytics/integrations.md.erb [hunk 5]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+
+### RSpec on Buildkite
+
+The following variables are set automatically when the RSpec integration is running on Buildkite:
+
+| Variable                       | Description and source                                                                   |
+|--------------------------------|------------------------------------------------------------------------------------------|
+| `BUILDKITE_ANALYTICS_KEY`      | Unique key for the build that initiated the Test Analytics run<br />`BUILDKITE_BUILD_ID` |
+| `BUILDKITE_ANALYTICS_URL`      | URL that links to the build <br /> `BUILDKITE_BUILD_URL`                                 |
+| `BUILDKITE_ANALYTICS_BRANCH`   | the branch or ref that this run is for<br />`BUILDKITE_BRANCH`                           |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:363]: ref - reference? `reference for this run` maybe?

File: pages/test_analytics/integrations.md.erb [hunk 6]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `BUILDKITE_ANANLYTICS_MESSAGE` | the commit message for the head of the branch<br />`BUILDKITE_JOB_ID`                    |
+
+
+### RSpec on CircleCI
+
+The following variables are set automatically when the RSpec integration is running on CircleCI:
+
+| Variable     | Description and source                                                                                          |
+|--------------|-----------------------------------------------------------------------------------------------------------------|
+| `key`        | Unique key for the build that initiated the Test Analytics run<br />`CIRCLE_WORKFLOW_ID` and `CIRCLE_BUILD_NUM` |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:376]:
```suggestion
| `key`        | a unique key for the build that initiated the Test Analytics run<br />`CIRCLE_WORKFLOW_ID` and `CIRCLE_BUILD_NUM` |
```

File: pages/test_analytics/integrations.md.erb [hunk 7]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `url`        | a URL that links to the build <br /> `CIRCLE_BUILD_URL`                                                         |
+| `branch`     | the branch or ref that this run is for<br />`CIRCLE_BRANCH`                                                     |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:378]: Same as for L363

File: pages/test_analytics/integrations.md.erb [hunk 8]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `commit_sha` | the commit sha for the head of the branch<br />`CIRCLE_SHA1`                                                    |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:379]:
```suggestion
| `commit_sha` | the commit SHA for the head of the branch<br />`CIRCLE_SHA1`                                                    |
```

File: pages/test_analytics/integrations.md.erb [hunk 9]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+
+To pass a human-readable commit message through to Test Analytics, set the `BUILDKITE_ANALYTICS_MESSAGE` variable.
+
+### RSpec on GitHub Actions
+
+The following variables are set automatically when the RSpec integration is running on GitHub Actions:
+
+| Variable     | Description and source                                                                                                               |
+|--------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| `key`        | Unique key for the build that initiated the Test Analytics run<br />`GITHUB_ACTION` and `GITHUB_RUN_NUMBER` and `GITHUB_RUN_ATTEMPT` |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:390]:
```suggestion
| `key`        | a unique key for the build that initiated the Test Analytics run<br />`GITHUB_ACTION` and `GITHUB_RUN_NUMBER` and `GITHUB_RUN_ATTEMPT` |
```

File: pages/test_analytics/integrations.md.erb [hunk 10]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `url`        | a URL that links to the build <br /> `GITHUB_REPOSITORY` and `GITHUB_RUN_ID`                                                         |
+| `branch`     | the branch or ref that this run is for<br />`GITHUB_REF`                                                                             |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:392]: Same as for L363

File: pages/test_analytics/integrations.md.erb [hunk 11]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `commit_sha` | the commit sha for the head of the branch<br />`GITHUB_REF`                                                                          |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:393]:
```suggestion
| `commit_sha` | the commit SHA for the head of the branch<br />`GITHUB_REF`                                                                          |
```

File: pages/test_analytics/integrations.md.erb [hunk 12]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+
+To pass a human-readable commit message through to Test Analytics, set the `BUILDKITE_ANALYTICS_MESSAGE` variable.
+
+### RSpec on other CI providers
+
+If you're using other CI providers, you need to configure these environment variables yourself. The only required environment variable is the unique key for the build that initiated the Test Analytics run (`BUILDKITE_ANALYTICS_KEY`), but some features of Test Analytics aren't available if the other environment variables are not set:
+
+| Variable                       | Description and source                                         |
+|--------------------------------|----------------------------------------------------------------|
+| `BUILDKITE_ANALYTICS_KEY`      | Unique key for the build that initiated the Test Analytics run |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:404]:
```suggestion
| `BUILDKITE_ANALYTICS_KEY`      | a unique key for the build that initiated the Test Analytics run |
```

File: pages/test_analytics/integrations.md.erb [hunk 13]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `BUILDKITE_ANALYTICS_URL`      | URL that links to the build                                    |
+| `BUILDKITE_ANALYTICS_BRANCH`   | the branch or ref that this run is for                         |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:406]: Same as for L363

File: pages/test_analytics/integrations.md.erb [hunk 14]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `BUILDKITE_ANALYTICS_SHA`      | the commit sha for the head of the branch                      |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:407]:
```suggestion
| `BUILDKITE_ANALYTICS_SHA`      | the commit SHA for the head of the branch                      |
```

File: pages/test_analytics/integrations.md.erb [hunk 15]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+
+### Other CI providers and integrations
+
+If you're using other tools, such as the [JUnit](#importing-junit-xml) or [JSON](#importing-json) importers, you need to configure the variables in `run_env` yourself.
+
+The only required variable is `key`, but some features of Test Analytics aren't available if the other variables are not set:
+
+| Variable     | Description                                                    |
+|--------------|----------------------------------------------------------------|
+| `key`        | Unique key for the build that initiated the Test Analytics run |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:421]:
```suggestion
| `key`        | a unique key for the build that initiated the Test Analytics run |
```

File: pages/test_analytics/integrations.md.erb [hunk 16]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `url`        | URL that links to the build                                    |
+| `branch`     | the branch or ref that this run is for                         |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:423]: Same as L363

File: pages/test_analytics/integrations.md.erb [hunk 17]
```diff
@@ -376,6 +336,97 @@ The JSON format is not yet documented, but the following example is what is prod
...
+| `commit_sha` | the commit sha for the head of the branch                      |
```
Comment (karensawrey) [inline pages/test_analytics/integrations.md.erb:424]:
```suggestion
| `commit_sha` | the commit SHA for the head of the branch                      |
```

**Review by karensawrey — APPROVED** [review]

File: pages/test_analytics/integrations.md.erb [hunk 18]
```diff
@@ -49,55 +49,20 @@ If you're already using RSpec for your tests, add the `rspec-buildkite-analytics
...
-* `CI`
-* `key` (**required**)
-* `number`
-* `job_id`
-* `branch`
-* `commit_sha`
-* `message`
-* `url`
+* if you're making the request in a Buildkite pipeline, you can use the Buildkite environment variables to set the metadata automatically, like in the following example:
+* if you're making it elsewhere, you'll need to [set them yourself](#environment-variables-other-ci-providers-and-integrations)
```
[the remark this answers is not in the document]
Comment (plaindocs) [reply pages/test_analytics/integrations.md.erb]: YES!

**Comment by lachlanhardy** [comment]
Woot!

**Pull request merged**